NuRadioReco/modules/RNO_G/stationDeepCRVariables.py

```diff
@@ -0,0 +1,149 @@
+from NuRadioReco.framework.parameters import channelParameters as chp, stationParametersRNOG as stpRNOG
+import matplotlib
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import signal
+from scipy.signal import hilbert
+from scipy.ndimage import maximum_filter1d, minimum_filter1d
+from NuRadioReco.modules.base.module import register_run
+import NuRadioReco.modules.channelSignalReconstructor
+from NuRadioReco.utilities import trace_utilities
+
+
+class stationDeepCRVariables:
+    """
+    Module that calculates some variables for a Linear Discriminant Analysis of a specific event and stores them at the station level.
+
+
+    """
+
+    def __init__(self):
+        pass
+
+    def begin(self, coincidence_window_size = 6, pad_length = 500, channel_ids = [0,1,2,3]):
+        """
+        Parameters
+        ----------
+
+        coincidence_window_size : int (default: 6)
+            Window size used for calculating the maximum peak to peak amplitude
+
+        pad_length : int (default 500)
+            Padding length used for calculating the coherent sum
+
+        channel_ids : array of int (default: [0,1,2,3])
+            Channels for which to calculate the variables
+
+        """
+        self.__coincidence_window_size = coincidence_window_size
+        self.__pad_length = pad_length
+        self.__channel_ids = channel_ids
+
+
+    @register_run()
+    def run(self, event, station, detector, ref_ch_id=0):
+
+        """
+        Calculate LDA variables and add to the station object.
+
+        Parameters
+        ----------
+
+        event: Event object
+            The event for which the LDA variables should be calculated
+            
+        station: Station object
+            The station for which the LDA variables should be calculated
+            
+        detector: Detector object
+            The detector description
+
+        ref_ch_id: int
+            reference channel for the coherent sum
+
+        """
+        station.set_parameter(stpRNOG.avg_ch_snr, self.avg_ch_snr(event, station, detector))
+        sum_chan=self.coherent_sum(event, station, station.get_channel(ref_ch_id))
+        station.set_parameter(stpRNOG.coherent_snr, self.coherent_snr(sum_chan))
+        return
+        
+    def end(self):
+        pass
+
+    def avg_ch_snr(self, event, station, detector):
+        snrs = []
+        for channel in station.iter_channels(use_channels=self.__channel_ids):
+            snrs.append(channel[chp.SNR]['peak_2_peak_amplitude_split_noise_rms'])
+        avg_snr = np.mean(snrs)
+        return avg_snr
```

Comment:
I do not think we should store this as parameter if it is a simple mean of channels which can easily be calculated from the nur files only using what is already in the parameter storage.